pandas/core/arrays/datetimelike.py

```diff
@@ -1082,26 +1087,123 @@ def _cmp_method(self, other, op):
     __divmod__ = make_invalid_op("__divmod__")
     __rdivmod__ = make_invalid_op("__rdivmod__")
 
+    @final
     def _add_datetimelike_scalar(self, other):
         # Overridden by TimedeltaArray
-        raise TypeError(f"cannot add {type(self).__name__} and {type(other).__name__}")
+        if not is_timedelta64_dtype(self.dtype):
+            raise TypeError(
+                f"cannot add {type(self).__name__} and {type(other).__name__}"
+            )
 
-    _add_datetime_arraylike = _add_datetimelike_scalar
+        from pandas.core.arrays import DatetimeArray
 
-    def _sub_datetimelike_scalar(self, other):
-        # Overridden by DatetimeArray
         assert other is not NaT
-        raise TypeError(f"cannot subtract a datelike from a {type(self).__name__}")
+        other = Timestamp(other)
+        if other is NaT:
+            # In this case we specifically interpret NaT as a datetime, not
+            # the timedelta interpretation we would get by returning self + NaT
+            result = self.asi8.view("m8[ms]") + NaT.to_datetime64()
+            return DatetimeArray(result)
+
+        i8 = self.asi8
+        result = checked_add_with_arr(i8, other.value, arr_mask=self._isnan)
+        result = self._maybe_mask_results(result)
+        dtype = DatetimeTZDtype(tz=other.tz) if other.tz else DT64NS_DTYPE
```

Comment:
Even with reso support, `DT64NS_DTYPE` is okay since ns is the default?

Reply:
when the time comes we'll change this to inherit self._reso

Reply:
Worth a TODO or in your local notes?

Reply:
not really.  I'll be grepping for DT64NS_DTYPE when the time comes anyway